walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,147 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Iterator;
+import java.util.Set;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+    
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        ArrayList<TimeRange> timesWithOptionals = new ArrayList<>();
+        ArrayList<TimeRange> timesNoOptionals = new ArrayList<>();
+        timesWithOptionals.add(TimeRange.WHOLE_DAY);
+        timesNoOptionals.add(TimeRange.WHOLE_DAY);
+
+        if (request.getDuration() > (24 * 60)) {
+            return new ArrayList<>();
+        } else if (events.size() == 0) {
+            return timesWithOptionals;
+        }
+
+        Collection<String> requestersWithOptionals = new ArrayList<>();
+        requestersWithOptionals.addAll(request.getAttendees());
+        requestersWithOptionals.addAll(request.getOptionalAttendees());
+        Collection<String> requestersNoOptionals = new ArrayList<>();
+        requestersNoOptionals.addAll(request.getAttendees());
+
+        ArrayList<Event> eventsWithOptionals = new ArrayList<>(events);
```

Comment:
This is a rather large chunk of code and hard to read. Could you separate this out into different helper method that address the different cases?

Reply:
Thank you for this comment, I totally forgot about that best coding practice and as you saw was doing for loops for two iterations